plugin/src/software/aws/toolkits/eclipse/amazonq/util/UpdateUtils.java

```diff
@@ -0,0 +1,123 @@
+
+        return remoteVersionIsGreater(remoteVersion, localVersion) && shouldShowNotification;
+    }
+
+    public void checkForUpdate() {
+        if (newUpdateAvailable()) {
+            showNotification();
+        }
+    }
+
+    private Version fetchRemoteArtifactVersion(final String repositoryUrl) {
+        HttpClient connection = HttpClientFactory.getInstance();
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(repositoryUrl))
+            .timeout(Duration.ofSeconds(5))
+            .GET()
+            .build();
+
+            HttpResponse<InputStream> response = connection.send(request,
+            HttpResponse.BodyHandlers.ofInputStream());
+
+            // handle response codes
+            if (response.statusCode() != HttpURLConnection.HTTP_OK) {
+                throw new AmazonQPluginException("HTTP request failed with response code: " + response.statusCode());
+            }
+
+            // process XZ content from input stream
+            try (InputStream inputStream = response.body();
+                 XZInputStream xzis = new XZInputStream(inputStream);
+                 BufferedReader reader = new BufferedReader(new InputStreamReader(xzis))) {
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (line.contains("<artifact classifier=\"osgi.bundle\"")) {
+                        int versionStart = line.indexOf("version=\"") + 9;
+                        int versionEnd = line.indexOf("\"", versionStart);
+                        String fullVersion = line.substring(versionStart, versionEnd);
+                        return ArtifactUtils.parseVersion(fullVersion.substring(0, fullVersion.lastIndexOf(".")));
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private void showNotification() {
+        Display.getDefault().asyncExec(() -> {
+            AbstractNotificationPopup notification = new PersistentToolkitNotification(Display.getCurrent(),
+                    Constants.PLUGIN_UPDATE_NOTIFICATION_TITLE,
+                    String.format(Constants.PLUGIN_UPDATE_NOTIFICATION_BODY, remoteVersion.toString()),
+                    (selected) -> {
+                        if (selected) {
+                            Activator.getPluginStore().putObject(Constants.DO_NOT_SHOW_UPDATE_KEY, remoteVersion);
+                        } else {
+                            Activator.getPluginStore().remove(Constants.DO_NOT_SHOW_UPDATE_KEY);
+                        }
+                    });
+            notification.open();
+        });
+    }
+
+    private static boolean remoteVersionIsGreater(final Version remote, final Version local) {
+        return remote.compareTo(local) > 0;
+    }
```

Comment:
One side note - #329 adds support for full semantic versioning of artifacts, and a different dependency (Maven Artifact library) for version constructs. Once that is merged you can switch to using those to get the benefits of being able to diff across labeled versions. Depending on when things get merged that can also be a follow up improvement.